File: tubefold/output_language.py
from __future__ import annotations


DEFAULT_OUTPUT_LANGUAGE = "English"
MAX_OUTPUT_LANGUAGE_LENGTH = 60


def normalize_output_language(value: str | None) -> str:
    """Clean a user-provided output-language label.

    Collapses whitespace/newlines, trims, caps the length, and falls back to
    the default when empty. The value is inserted verbatim into the prompt, so
    keep it to a short single line.
    """
    if value is None:
        return DEFAULT_OUTPUT_LANGUAGE
    cleaned = " ".join(str(value).split())
    if not cleaned:
        return DEFAULT_OUTPUT_LANGUAGE
    if len(cleaned) > MAX_OUTPUT_LANGUAGE_LENGTH:
        cleaned = cleaned[:MAX_OUTPUT_LANGUAGE_LENGTH].rstrip()
    return cleaned

File: TubeFoldKit/Sources/TubeFoldKit/Core/OutputLanguage.swift
import Foundation

/// Output-language setting normalization.
public enum OutputLanguage {
    public static let defaultLanguage = "English"
    public static let maxLength = 60

    /// Clean a user-provided output-language label.
    ///
    /// Collapses whitespace/newlines, trims, caps the length, and falls back
    /// to the default when empty. The value is inserted verbatim into the
    /// prompt, so keep it to a short single line.
    public static func normalize(_ value: String?) -> String {
        guard let value else { return defaultLanguage }
        var cleaned = value
            .split(whereSeparator: \.isWhitespace)
            .joined(separator: " ")
        if cleaned.isEmpty {
            return defaultLanguage
        }
        if cleaned.count > maxLength {
            cleaned = String(cleaned.prefix(maxLength)).trimmingTrailing(charactersIn: " \t\n\r")
        }
        return cleaned
    }
}
